Add release notes for upcoming 1.5.1 release

garyo opened the pull request:
This needs to go out soon so I can update the Conancenter repo with Windows fixes (see also #224 , and please review that as well so I can merge it), and check out https://github.com/conan-io/conan-center-index/pull/26667 if you're interested in why the Windows stuff matters in the conanfile.

barretpj commented on release-notes.md:
```diff
@@ -19,14 +19,48 @@ This is version 1.6 of the OpenFX API.
 
 ## Fixes in OpenFX Version 1.6:
```
Should this still say 1.6 (here and above)?

garyo replied:
We've always had a placeholder for the next release... but we could keep "un-released" changes elsewhere if you think that would help. It would certainly make this cleaner.

barretpj reviewed (changes requested):
Feedback added

garyo commented:
Thanks for the comments, Phil! I addressed them above. I deliberately left the 1.6 section above, to serve as a place to collect things that will go into 1.6 (that's how we usually do it, though I agree it can be confusing—suggestions appreciated! I moved the SpatialAwareness prop into 1.5.1, which is why it looked like it was getting deleted.
